purchase_work_acceptance_invoice_plan/wizard/select_work_acceptance_invoice_plan_wizard.py

```diff
@@ -0,0 +1,31 @@
+
+# Copyright 2019 Ecosoft Co., Ltd. (http://ecosoft.co.th).
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, models, fields, api
+from odoo.exceptions import ValidationError
+
+
+class SelectWorkAcceptanceInvoicePlanWizard(models.TransientModel):
+    _name = "select.work.acceptance.invoice.plan.wizard"
+    _description = "Select Work Acceptance Invoice Plan Wizard"
+
+    wa_in_id = fields.Many2one(
+        comodel_name="purchase.invoice.plan",
+        string="Invoice Plan",
+        required=True,
+        domain=lambda self: self._get_domain_wa_in_id(),
+    )
+
+    def _get_domain_wa_in_id(self):
+        purchases = self.env["purchase.order"].browse(self._context.get("active_id"))
+        res = [("purchase_id", "in", purchases.ids), ("id", "not in", purchases.mapped("wa_ids.wa_in_id").ids)]
```

Comment:
```suggestion
        res = [("id", "not in", purchases.mapped("wa_ids.wa_in_id").ids)]
```
Since we test with "id", I think no need for "purchase_id".
Note: the logic might need to change a bit, as we don't remove wa_in_id from WA when cancel.